src/pages/Timetable/components/TimetableList/TimetableSettingModal/index.tsx

```diff
@@ -3,20 +3,49 @@ import { cn } from '@bcsdlab/utils';
 import { ReactComponent as CloseIcon } from 'assets/svg/close-icon-black.svg';
 import { ReactComponent as CheckedIcon } from 'assets/svg/checked-icon.svg';
 import { ReactComponent as NotCheckedIcon } from 'assets/svg/not-checked-icon.svg';
+import { TimetableFrameInfo } from 'api/timetable/entity';
+import useUpdateTimetableFrame from 'pages/Timetable/hooks/useUpdateTimetableFrame';
+import useDeleteTimetableFrame from 'pages/Timetable/hooks/useDeleteTimetableFrame';
+import useTokenState from 'utils/hooks/useTokenState';
+import { useSemester } from 'utils/zustand/semester';
+import useToast from 'components/common/Toast/useToast';
 import styles from './TimetableSettingModal.module.scss';
 
 export interface TimetableSettingModalProps {
+  focusFrame: TimetableFrameInfo
+  setFocusFrame: (frame: TimetableFrameInfo) => void
```

Comment:
```suggestion
  toggleIsChecked: () => void
```
모달 내에 있는 함수(`toggleIsChecked`)를 인자로 넘기는 것도 괜찮은 것 같아요! 나중에 `setState`를 너무 많이 사용하게 되면 의도치 않은 상태를 변경시킬 수 있고, 재사용성 측면에서도 훨씬 좋을 것 같아요!

Reply:
모달 내에서 focusFrame의 is_main 변경 외에 이름 변경도 있어서 그대로 `setState`를 넘겨도 될 것 같다고 생각합니다..! is_main만 변경했다면 `toggleIsChecked`를 인자로 넘기는 방법이 좋겠네요! 새로운 접근 방식 배워갑니다!!